### DISCUSSION TITLE
HDD Temps inside docker container?

### DISCUSSION THREAD

**Discussion by jtpavlock** [nicolargo/glances#3224] [General]
#### Description

Is there anyway currently to get hdd temps while glances is running inside of a docker container?

Maybe I'm bad at searching, but I couldn't find anything / could use some help if this is possible.

#### Versions

* Glances (glances -V): current docker container image
* PSutil (glances -V):
* Operating System (lsb_release -a): Debian 9.2

**Comment by nicolargo** [comment]
Not an easy trick...

Glances uses a socket connection to the HDDTemp daemon running on the host (by default on 127.0.0.1:7634). So you had to : 
1) configure your HDDTemp daemon to listen to all interfaces
2) find a way to map this host local port inside the Glances container

If a contributor has an idea...

**Comment by dalberto** [comment]
So I think this should be possible:

1. `hddtemp` has `--listen` option that would allow listening on all interfaces.
2. It is possible for a container to access services on the hosts but the address is different based on whether Docker is running natively or not (i.e. Docker for Mac / Windows). This seems like a simple wrapper script that can handle the differences though: https://github.com/bufferings/docker-access-host

**Comment by wxhbts** [comment]
docker  increase --net=host

docker run -d --restart="always" -p 61208-61209:61208-61209 -e GLANCES_OPT="-w" -v /var/run/docker.sock:/var/run/docker.sock:ro --pid host --net=host --name glances nicolargo/glances:latest-full
